"Model not available" error on non-Gradle project

### Expected Behavior
No error at all. I would even expect Buildship to do nothing on a non-Gradle project.

### Current Behavior
Recently I started to get an error when building some Groovy projects (through "clean project" or through a change in a Groovy source file). These projects are NOT Gradle projects.
This is a screenshot:
![immagine](https://user-images.githubusercontent.com/1742494/79312648-7bfa8f80-7eff-11ea-829e-6bc920b8551a.png)

And this is the contents of the Eclipse error log:

```
eclipse.buildId=4.14.0.I20191210-0610
java.version=1.8.0_242
java.vendor=Private Build
BootLoader constants: OS=linux, ARCH=x86_64, WS=gtk, NL=it_IT
Command-line arguments:  -data file:/home/mauro/workspace/rolling/DCS-SHOP-trunk/ -os linux -ws gtk -arch x86_64

org.eclipse.core.resources
Error
Wed Apr 15 09:56:40 CEST 2020
Problems occurred when invoking code from plug-in: "org.eclipse.core.resources".

java.lang.RuntimeException: Failed to bootstrap GroovyClassLoaders for project 'TestGroovy'
	at org.codehaus.jdt.groovy.internal.compiler.GroovyClassLoaderFactory.getProjectGroovyClassLoaders(GroovyClassLoaderFactory.java:174)
	at org.codehaus.jdt.groovy.internal.compiler.GroovyClassLoaderFactory.getGroovyClassLoaders(GroovyClassLoaderFactory.java:115)
	at org.codehaus.jdt.groovy.internal.compiler.ast.GroovyParser.lambda$0(GroovyParser.java:120)
	at org.codehaus.jdt.groovy.internal.compiler.ast.GroovyParser.dietParse(GroovyParser.java:181)
	at org.codehaus.jdt.groovy.integration.internal.MultiplexingParser.dietParse(MultiplexingParser.java:45)
	at org.eclipse.jdt.internal.compiler.Compiler.internalBeginToCompile(Compiler.java:861)
	at org.eclipse.jdt.internal.compiler.Compiler.beginToCompile(Compiler.java:404)
	at org.eclipse.jdt.internal.compiler.Compiler.compile(Compiler.java:454)
	at org.eclipse.jdt.internal.compiler.Compiler.compile(Compiler.java:436)
	at org.eclipse.jdt.internal.core.builder.AbstractImageBuilder.compile(AbstractImageBuilder.java:410)
	at org.eclipse.jdt.internal.core.builder.BatchImageBuilder.compile(BatchImageBuilder.java:218)
	at org.eclipse.jdt.internal.core.builder.AbstractImageBuilder.compile(AbstractImageBuilder.java:342)
	at org.eclipse.jdt.internal.core.builder.BatchImageBuilder.build(BatchImageBuilder.java:80)
	at org.eclipse.jdt.internal.core.builder.JavaBuilder.buildAll(JavaBuilder.java:275)
	at org.eclipse.jdt.internal.core.builder.JavaBuilder.build(JavaBuilder.java:187)
	at org.eclipse.core.internal.events.BuildManager$2.run(BuildManager.java:833)
	at org.eclipse.core.runtime.SafeRunner.run(SafeRunner.java:45)
	at org.eclipse.core.internal.events.BuildManager.basicBuild(BuildManager.java:220)
	at org.eclipse.core.internal.events.BuildManager.basicBuild(BuildManager.java:263)
	at org.eclipse.core.internal.events.BuildManager$1.run(BuildManager.java:316)
	at org.eclipse.core.runtime.SafeRunner.run(SafeRunner.java:45)
	at org.eclipse.core.internal.events.BuildManager.basicBuild(BuildManager.java:319)
	at org.eclipse.core.internal.events.BuildManager.basicBuildLoop(BuildManager.java:371)
	at org.eclipse.core.internal.events.BuildManager.build(BuildManager.java:392)
	at org.eclipse.core.internal.events.AutoBuildJob.doBuild(AutoBuildJob.java:154)
	at org.eclipse.core.internal.events.AutoBuildJob.run(AutoBuildJob.java:244)
	at org.eclipse.core.internal.jobs.Worker.run(Worker.java:63)
Caused by: java.lang.RuntimeException: java.lang.reflect.InvocationTargetException
	at org.codehaus.jdt.groovy.internal.compiler.GroovyClassLoaderFactory.calculateClasspath(GroovyClassLoaderFactory.java:198)
	at org.codehaus.jdt.groovy.internal.compiler.GroovyClassLoaderFactory.lambda$2(GroovyClassLoaderFactory.java:151)
	at java.util.concurrent.ConcurrentHashMap.computeIfAbsent(ConcurrentHashMap.java:1660)
	at org.codehaus.jdt.groovy.internal.compiler.GroovyClassLoaderFactory.getProjectGroovyClassLoaders(GroovyClassLoaderFactory.java:149)
	... 26 more
Caused by: java.lang.reflect.InvocationTargetException
	at sun.reflect.GeneratedMethodAccessor47.invoke(Unknown Source)
	at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at java.lang.reflect.Method.invoke(Method.java:498)
	at org.eclipse.jdt.groovy.core.util.ReflectionUtils.throwableExecutePrivateMethod(ReflectionUtils.java:117)
	at org.codehaus.jdt.groovy.internal.compiler.GroovyClassLoaderFactory.resolveRuntimeClasspathEntry(GroovyClassLoaderFactory.java:204)
	at org.codehaus.jdt.groovy.internal.compiler.GroovyClassLoaderFactory.calculateClasspath(GroovyClassLoaderFactory.java:191)
	... 29 more
Caused by: org.eclipse.buildship.core.internal.GradlePluginsRuntimeException: Model not available for TestGroovy
	at org.eclipse.buildship.core.internal.workspace.GradleClasspathContainerRuntimeClasspathEntryResolver.resolveRuntimeClasspathEntry(GradleClasspathContainerRuntimeClasspathEntryResolver.java:77)
	at org.eclipse.buildship.core.internal.workspace.GradleClasspathContainerRuntimeClasspathEntryResolver.resolveRuntimeClasspathEntry(GradleClasspathContainerRuntimeClasspathEntryResolver.java:66)
	at org.eclipse.jdt.internal.launching.RuntimeClasspathEntryResolver.resolveRuntimeClasspathEntry(RuntimeClasspathEntryResolver.java:110)
	at org.eclipse.jdt.launching.JavaRuntime.resolveRuntimeClasspathEntry(JavaRuntime.java:1505)
	at org.eclipse.jdt.internal.launching.DefaultEntryResolver.resolveRuntimeClasspathEntry(DefaultEntryResolver.java:70)
	at org.eclipse.jdt.launching.JavaRuntime.resolveRuntimeClasspathEntry(JavaRuntime.java:1508)
	at org.eclipse.jdt.launching.JavaRuntime.resolveRuntimeClasspathEntry(JavaRuntime.java:1444)
	... 35 more
```
The mentioned `TestGroovy` project is an Eclipse Groovy project with neither a Gradle nature nor a `build.gradle` script.

This error was also reported to the Groovy Eclipse plugin issue tracker: https://github.com/groovy/groovy-eclipse/issues/1087

### Context
Just trying to perform an incremental or full build of a plain Eclipse Groovy project (not being a Gradle project).
Started to happen recently, although I can't say when. Perhaps from when I upgraded to Eclipse 2019-12, but I don't know for sure.

Current Buildship version is 3.1.3.v20191118-1057.

### Steps to Reproduce 
Create a simple Groovy project (not a Gradle project).
Restart your IDE.
Do Project | Clean... to clean that project.

I don't know if this can be reproduced consistently just after the first restart after creating the Groovy project. I'm pretty sure that upon creation, no error is given until you restart, because I know for sure that I used this `TestGroovy` project for a while, with no problems. Currently, I can reproduce this problem always, whenever I try yo clean this project or to save a change in any of the Groovy source files in it.
I was not able to find a way to fix this problem. I usually have to delete the affected project and re-create from scratch. Fortunately for me, this is happening on test projects.

### Your Environment
This happens on both a Windows and a Linux systems. They both use the same Eclipse, Groovy Eclipse plugin and Buildship versions.

--- a/org.eclipse.buildship.core/src/main/java/org/eclipse/buildship/core/internal/workspace/GradleClasspathContainerRuntimeClasspathEntryResolver.java
+++ b/org.eclipse.buildship.core/src/main/java/org/eclipse/buildship/core/internal/workspace/GradleClasspathContainerRuntimeClasspathEntryResolver.java
@@ -60,12 +60,12 @@ public final class GradleClasspathContainerRuntimeClasspathEntryResolver impleme
 
     @Override
     public IRuntimeClasspathEntry[] resolveRuntimeClasspathEntry(IRuntimeClasspathEntry entry, IJavaProject project) throws CoreException {
-        return resolveRuntimeClasspathEntry(entry, project, LaunchConfigurationScope.INCLUDE_ALL, false, hasModuleSupport());
+        return resolveRuntimeClasspathEntry(entry, entry.getJavaProject(), LaunchConfigurationScope.INCLUDE_ALL, false, hasModuleSupport());
     }
 
     // @Override commented out as this method doesn't exist older Eclipse versions
     public IRuntimeClasspathEntry[] resolveRuntimeClasspathEntry(IRuntimeClasspathEntry entry, IJavaProject project, boolean excludeTestCode) throws CoreException {
-        return resolveRuntimeClasspathEntry(entry, project, LaunchConfigurationScope.INCLUDE_ALL, excludeTestCode, hasModuleSupport());
+        return resolveRuntimeClasspathEntry(entry, entry.getJavaProject(), LaunchConfigurationScope.INCLUDE_ALL, excludeTestCode, hasModuleSupport());
     }
 
     private static boolean hasModuleSupport() {
@@ -77,7 +77,6 @@ public final class GradleClasspathContainerRuntimeClasspathEntryResolver impleme
         if (entry.getType() != IRuntimeClasspathEntry.CONTAINER || !entry.getPath().equals(GradleClasspathContainer.CONTAINER_PATH)) {
             return new IRuntimeClasspathEntry[0];
         }
-
         PersistentModel model = CorePlugin.modelPersistence().loadModel(project.getProject());
         if (!model.isPresent()) {
             throw new GradlePluginsRuntimeException("Model not available for " + project.getProject().getName());

